### PR TITLE
WAITP-1293: Refactor tupaia-web for flat entities data

### PR DESCRIPTION
**Pull request by tcaiger**
### Changes:
The context of this refactor is that in [waitp-1293-entity-route-better-filtering](https://github.com/beyondessential/tupaia/tree/waitp-1293-entity-route-better-filtering) the response from the entities api endpoint was refactored to return a flat list of entities and so the front end has to be refactored to match.

- Entity Menu
- Entity Polygons

### CODE REVIEW

File: packages/tupaia-web/src/api/queries/useEntity.ts [hunk 1]
```diff
@@ -4,34 +4,32 @@
...
-  countryHierarchy: [],
-};
+import { useParams } from 'react-router-dom';
+import { Entity } from '../../types';
+import { get } from '../api';
+import { DEFAULT_BOUNDS } from '../../constants';
 
 export const useEntity = (entityCode?: string) => {
+  //  Todo: use entity endpoint when it's done and remove project code
+  const { projectCode } = useParams();
```
Comment (tcaiger) [inline packages/tupaia-web/src/api/queries/useEntity.ts:14]: There is a ticket coming soon for a single entity route but this is useful for testing.

File: packages/tupaia-web/src/features/MapOverlays/PolygonLayer.tsx [hunk 2]
```diff
@@ -23,22 +24,30 @@ const ChildEntities = ({ entities }: { entities: EntityResponse['children'] }) =
   );
 };
 
-const SiblingEntities = ({ entity }: { entity: EntityResponse }) => {
+const SiblingEntities = ({
+  parentEntityCode,
+  activeEntityCode,
+}: {
+  parentEntityCode: Entity['code'];
+  activeEntityCode: Entity['code'];
```
Comment (acdunham) [inline packages/tupaia-web/src/features/MapOverlays/PolygonLayer.tsx:32]: I believe we have a helper type of `EntityCode` which you could also use here if you wanted, but it's not imperative.

I also tend to like to go by the rule of 'if a component has more than 1 prop, create an interface for it' so if you wanted to, you could add one here.

File: packages/tupaia-web/src/features/MapOverlays/PolygonLayer.tsx [hunk 3]
```diff
@@ -63,16 +72,32 @@ const ActiveEntity = ({ entity }: { entity: EntityResponse }) => {
...
 
-export const PolygonLayer = ({ entityData }: { entityData: EntityResponse }) => {
-  if (!entityData) {
+export const PolygonLayer = ({
+  entities,
+  entityCode,
+}: {
+  entities: EntityResponse[];
+  entityCode: string;
+}) => {
```
Comment (acdunham) [inline packages/tupaia-web/src/features/MapOverlays/PolygonLayer.tsx:81]: Same here, you could maybe add an interface for this one

File: packages/tupaia-web/src/features/MapOverlays/PolygonLayer.tsx [hunk 4]
```diff
@@ -23,22 +24,30 @@ const ChildEntities = ({ entities }: { entities: EntityResponse['children'] }) =
...
   );
 
-  if (isLoading || !siblingEntities) {
+  if (isLoading || !siblingEntities || siblingEntities.length === 0) {
     return null;
   }
 
-  const children = siblingEntities?.children?.filter(e => e.code !== entity.code) || [];
+  const children = siblingEntities.filter(
+    (entity: EntityResponse) => entity.code !== activeEntityCode,
```
Comment (acdunham) [inline packages/tupaia-web/src/features/MapOverlays/PolygonLayer.tsx:45]: I wonder if the types need to be added here, or if having them in the `useEntitiesWithLocation` would solve any TS errors with that? Might save us writing it out each time

**Review by acdunham — APPROVED** [review]
Looks good!  A couple of questions/comments, but otherwise pre-approving

File: packages/tupaia-web/src/features/MapOverlays/PolygonLayer.tsx (hunk 4 above)
Comment (tcaiger) [reply packages/tupaia-web/src/features/MapOverlays/PolygonLayer.tsx:45]: oh yes you are correct. I didn't need them here. I must have added them in the wrong order. thanks

**Pull request merged**